chain/chunks/src/lib.rs

```diff
@@ -612,10 +612,8 @@ impl ShardsManager {
                 continue;
             }
 
-            let fetch_from = if request_from_archival {
-                shard_representative_target.clone()
-            } else if we_own_part {
-                // If missing own part, request it from the chunk producer / node tracking shard
+            // If missing own part, request it from the chunk producer / node tracking shard
+            let fetch_from = if request_from_archival || we_own_part {
```

Comment:
I think this change should be undone and `#[allowed]`. See https://github.com/rust-lang/rust-clippy/issues/3770 / https://github.com/rust-lang/rust-clippy/issues/5940 for some context too.

Reply:
added `allow` and a comment referencing the issue